### PR TITLE
Serialize parallel downloads, fixes #1774

### LINKED ISSUE #1774: Long sessions playback is sometimes interrupted
**What happened**:

Playback of the long sessions is sometimes incomplete.

**What you expected to happen**:

Playback recorded session to the end

**How to reproduce it (as minimally and precisely as possible)**:

This issue is for teleport 2.6.0-alpha.0

Setup the teleport with dynamodb backend and s3 upload:

```yaml
# Single-node Teleport cluster called "one" (runs all 3 roles: proxy, auth and node)
teleport:
  nodename: one
  advertise_ip: 172.10.1.1
  log:
    output: /var/lib/teleport/teleport.log
    severity: INFO

  data_dir: /var/lib/teleport
  storage:
      type: dynamodb
      table_name: test_grv8
      region: us-west-1
      audit_table_name: test_grv8_events
      audit_sessions_uri: s3://testgrv8records
```

Open the terminal session in the UI. 

```bash
# run top for a few seconds
top
# produce significant output
ls -lR /
# print something and exit
echo howdy
exit
```

Then try to playback the session. For the first playback the session will be played back to about 70%
and stop. Second playback will work.

I've checked on backend and it's the same set of files, there is no eventual consistency as in 2.6.0 the session is fully downloaded on replay and is played from the same tarball.

That's why I'm assuming there is UI somehow involved.

**Environment**:
- Teleport version (use `teleport version`):

2.6.0-alpha.0

- Tsh version (use `tsh version`):
- OS (e.g. from /etc/os-release):

**Browser environment**

- Browser Version (for UI-related issues):
- Install tools:
- Others:

**Relevant Debug Logs If Applicable**

- tsh --debug
- teleport --debug

### PR DESCRIPTION
**Pull request by klizhentas**
In case if multiple requests to get session
event data were issued to the auth server
at the same time, multiple download requests
were originated, and sometimes partial data
was returned.

This commit serializes downloads of the session
in the context of the same auth server.

### CODE REVIEW

**Comment by klizhentas** [comment]
retest this please

**Review by russjones — APPROVED** [review]

**Comment by klizhentas** [comment]
retest this please

File: lib/events/auditlog.go [hunk 1]
```diff
@@ -552,12 +610,25 @@ func (l *AuditLog) downloadSession(namespace string, sid session.ID) error {
...
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, fileName := range idx.chunkFileNames() {
+		reader, err := l.unpackFile(fileName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if err := reader.Close(); err != nil {
+			l.Warningf("Failed to close file: %v", err)
```
Comment (alex-kovoy) [inline lib/events/auditlog.go]: missing . at the end

**Review by alex-kovoy — APPROVED** [review]

**Comment by klizhentas** [comment]
retest this please

**Pull request merged**